### ISSUE TITLE
CI tests broken on OS X

### ISSUE THREAD

**Issue by thomas-moulard** [ros-tooling/libstatistics_collector#7]
## Description

Tests broken on OS X:

```
2020-04-20T19:24:22.8349430Z [14.088s] The following tests FAILED:
2020-04-20T19:24:22.8349850Z [14.088s] 	  1 - test_collector (Failed)
2020-04-20T19:24:22.8350270Z [14.088s] 	  2 - test_moving_average_statistics (Failed)
2020-04-20T19:24:22.8350680Z [14.088s] 	  3 - test_received_message_period (Failed)
2020-04-20T19:24:22.8351080Z [14.088s] 	  4 - test_received_message_age (Failed)
2020-04-20T19:24:22.8351180Z [14.133s] Errors while running CTest
```

https://pipelines.actions.githubusercontent.com/770SLwBj3ET8NDQCRWFqpZmRWXkXzxULg9phWHBbZVu2jXGrW5/_apis/pipelines/1/runs/126/signedlogcontent/6?urlExpires=2020-04-20T19%3A53%3A59.6774078Z&urlSigningMethod=HMACV1&urlSignature=kOrEbPjDAN8O2b10B43olmT0lAJgPn4IaRNqs1GPPsI%3D

## To Reproduce

CI Run https://github.com/ros-tooling/libstatistics_collector/actions

## System (please complete the following information)
 - OS: OS X
 - ROS 2 Distro: master

## Additional context

N/A

**Comment by dabonnie** [comment]
I went down the path of trying to lookup the error

```
dyld: Library not loaded: @rpath/libspdlog.1.dylib
```

and found a few threads. [This](https://github.com/ros2/ros2/issues/391) was interesting and I'm not familiar with the image, but wondering if [SIP](https://developer.apple.com/library/archive/documentation/Security/Conceptual/System_Integrity_Protection_Guide/ConfiguringSystemIntegrityProtection/ConfiguringSystemIntegrityProtection.html) is enabled? Either way I found a few missing brew dependencies so will open a ticket and PR against `setup-ros`.

**Comment by dabonnie** [comment]
Using the draft PR #9 I was able to verify that [SIP](https://developer.apple.com/library/archive/documentation/Security/Conceptual/System_Integrity_Protection_Guide/ConfiguringSystemIntegrityProtection/ConfiguringSystemIntegrityProtection.html) is enabled on OSX

```
Mac-1557:libstatistics_collector runner$ csrutil status
System Integrity Protection status: enabled.
```

Will file a ticket against `setup-ros`.

**Comment by dabonnie** [comment]
This appears to be fixed by https://github.com/ros-tooling/libstatistics_collector/pull/10 and is unrelated to ros-tooling/setup-ros#128.

**Comment by dabonnie** [comment]
Closing given green CI and the fix #10 via @thomas-moulard.
